test/box-luatest/gh_4693_formats_for_standalone_tuples_test.lua

```diff
@@ -0,0 +1,136 @@
+    cg.server:exec(function()
+        box.error.injection.set('ERRINJ_TUPLE_FORMAT_COUNT', -1)
+    end)
+end)
+
+-- Checks that `box.tuple.new` with format option works as expected.
+g.test_box_tuple_new_with_format = function(cg)
+    cg.server:exec(function()
+        local f = {{name = 'field', type = 'number'}}
+        local options = {format = f}
+        local tuple = box.tuple.new({1}, options)
+        -- Checks that options table is not changed by box.tuple.new.
+        t.assert_equals(options, {format = f})
+        t.assert_equals(tuple:tomap{names_only = true}, {field = 1})
+        tuple = box.tuple.new(1, options)
+        t.assert_equals(tuple:tomap{names_only = true}, {field = 1})
+        tuple = box.tuple.new(box.tuple.new(1, options), options)
+        t.assert_equals(tuple:tomap{names_only = true}, {field = 1})
+        tuple = box.tuple.new({1, 'str'}, {format = f})
+        t.assert_equals(tuple:tomap{names_only = true}, {field = 1})
+        local err_msg = "Tuple field 1 (field) type does not match one " ..
+                        "required by operation: expected number, got string"
+        t.assert_error_msg_content_equals(err_msg, function()
+            box.tuple.new({'str'}, {format = f})
+        end)
+        err_msg = "Tuple field 1 (field) required by space format is missing"
+        t.assert_error_msg_content_equals(err_msg, function()
+            box.tuple.new({}, {format = f})
+        end)
+        f = {{name = 'field', type = 'number', is_nullable = true}}
+        tuple = box.tuple.new({}, {format = f})
+        t.assert_equals(tuple:tomap{names_only = true}, {})
+        f = box.tuple.format.new{{'field', 'number'}}
+        tuple = box.tuple.new({1}, {format = f})
+        t.assert_equals(tuple:tomap{names_only = true}, {field = 1})
+        tuple = box.tuple.new({1, 'str'}, {format = f})
+        t.assert_equals(tuple:tomap{names_only = true}, {field = 1})
+        err_msg = "box.tuple.new(): second argument (options) must be a table"
+        t.assert_error_msg_contains(err_msg, function()
+            box.tuple.new({'str'}, 'fmt')
+        end)
+        err_msg = "Usage: box.tuple.new({tuple-fields}, {options})"
+        t.assert_error_msg_contains(err_msg, function()
+            box.tuple.new({'str'}, {format = f}, 3)
+        end)
+
+        -- Checks that check constraint is disabled.
+        box.schema.func.create('ck', {is_deterministic = true,
+                                      body = "function () return false end"})
+        box.tuple.new({0}, {format = {{'field', constraint = 'ck'}}})
+        box.func.ck:drop()
+
+        -- Checks that foreign key constraint is disabled.
+        box.schema.space.create('s')
+        box.tuple.new({0}, {format =
+            {{'field', foreign_key = {space = 's', field = 1}}}})
+        box.space.s:drop()
+        box.tuple.format.new{{'field', foreign_key = {field = 1}}}
+        box.tuple.format.new{{'field', foreign_key = {fk = {field = 1}}}}
```

Comment:
Please also try to create a foreign key with a non-existing space name and id.

Reply:
Added a test case.